sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2876,6 +2877,33 @@ class HiveDDLSuite
     }
   }
 
+  test("SPARK-33865: Hive DDL with avro should check col name") {
+    withTable("t1") {
+      withView("v") {
+        spark.range(1).createTempView("v")
+
+        val e1 = intercept[SchemaParseException] {
```

Comment:
Shall we unify the exception type? It's weird if different formats use different exception types.

Reply:
> Shall we unify the exception type? It's weird if different formats use different exception types.

Yea, we can do this  since we.a unify spark's exception. add a sub-task https://issues.apache.org/jira/browse/SPARK-34402